aten/src/ATen/NamedTensorUtils.cpp

```diff
@@ -248,17 +248,63 @@ void propagate_names(TensorImpl* result, TensorImpl* src) {
   propagate_names(result, impl::get_names(src));
 }
 
-static std::vector<Dimname> compute_mm_outnames(
-    optional<DimnameList> mat1_names,
-    optional<DimnameList> mat2_names) {
-  std::vector<Dimname> mm_outnames(2, Dimname::wildcard());
-  if (mat1_names) {
-    mm_outnames[0] = (*mat1_names)[0];
+static optional<std::vector<Dimname>> compute_dot_product_outnames(
+    optional<DimnameList> tensor_names,
+    int64_t tensor_dotted_dim,
+    int64_t tensor_ndim,
+    optional<DimnameList> other_names,
+    int64_t other_dotted_dim,
+    int64_t other_ndim) {
+  int64_t num_outnames = tensor_ndim + other_ndim - 2;
+  if (num_outnames == 0) {
+    return nullopt;
+  }
+  std::vector<Dimname> outnames(num_outnames, Dimname::wildcard());
+  int64_t index = 0;
+  if (tensor_names) {
+    for (int64_t j = 0; j < tensor_names->size(); ++j) {
+      if (j == tensor_dotted_dim) continue;
+      outnames[index++] = (*tensor_names)[j];
+    }
+  }
+  index = tensor_ndim - 1;
+  if (other_names) {
+    for (int64_t j = 0; j < other_names->size(); ++j) {
+      if (j == other_dotted_dim) continue;
+      outnames[index++] = (*other_names)[j];
+    }
+  }
+  return outnames;
+}
+
+static optional<DimnameList> to_opt_dimnames(
+    const optional<std::vector<Dimname>>& names) {
+  if (names) {
+    return *names;
   }
-  if (mat2_names) {
-    mm_outnames[1] = (*mat2_names)[1];
+  return nullopt;
+}
+
+void propagate_names_for_addmv(
+    TensorImpl* result,
+    TensorImpl* mat,
+    TensorImpl* vec,
+    TensorImpl* bias) {
+  if (!impl::has_names(result) && !impl::has_names(mat) &&
+      !impl::has_names(vec) && !impl::has_names(bias)) {
+    return;
   }
-  return mm_outnames;
+  auto mv_outnames = compute_dot_product_outnames(
```

Comment:
this function has a lot of parameters, which can make it harder to read. 3 of them are passed as hard-coded 1, 2, 0 in all uses in this review. are there cases where they will be different or can we simplify this code?

Reply:
I found other uses in another review/stack ... but still seems like there might be some simplification that can be made here. e.g. is `other_dotted_dim` always going to be 0? is other_ndim just the number of dims in `other`?

Reply:
> e.g. is other_dotted_dim always going to be 0

Probably.

> other_ndim just the number of dims in other

Only if `other` isn't nullopt. If `other` is nullopt then we don't know what's going on.

Reply:
I've changed it so that there are only 4 parameters instead of 6. I don't want to hardcode `other_dotted_dim` to 0, but I can do that if you think it would be better.

Reply:
It's still not clear to me why not part of the implementation, and I think that could be confusing to anyone else who might want to call the function.

If it appears that it should always be zero, but it's a function parameter, personally I'd assume I miss-understood the meaning of the parameter. I would probably spend a lot of time trying to figure out what the other valid values would be/why before passing a zero. If there's a reason to keep it, maybe a comment would help explain why it's there or when something other than zero would make sense.

Reply:
For something like einsum, we'd use a dimension other than zero, however the function in its current state isn't sufficient for einsum. When one is doing tensor contraction (a generalized matrix multiply), one can pick one dimension from each input tensor to contract together. That dimension just happens to always be the last dimension of the first tensor and the zeroth dimension of the second tensor, but it isn't in the case of `einsum`.

Reply:
I'll go toss what I wrote in that comment into a comment in the code

Reply:
that makes sense, I think an explanation in comments would be fine then